contracts/external/traders/CanonicalOrders.sol

```diff
@@ -0,0 +1,842 @@
+        });
+    }
+
+    /**
+     * Increases the stored filled amount of the order by fillAmount.
+     * Returns the new total filled amount.
+     */
+    function updateFilledAmount(
+        OrderInfo memory orderInfo,
+        uint256 fillAmount
+    )
+        private
+    {
+        uint256 oldFilledAmount = g_filledAmount[orderInfo.orderHash];
+        uint256 totalFilledAmount = oldFilledAmount.add(fillAmount);
+        Require.that(
+            totalFilledAmount <= orderInfo.order.amount,
+            FILE,
+            "Cannot overfill order",
+            orderInfo.orderHash,
+            oldFilledAmount,
+            fillAmount
+        );
+
+        g_filledAmount[orderInfo.orderHash] = totalFilledAmount;
+
+        emit LogCanonicalOrderFilled(
+            orderInfo.orderHash,
+            orderInfo.order.makerAccountOwner,
+            fillAmount,
+            totalFilledAmount,
+            orderInfo.tradeArgs.price,
+            orderInfo.tradeArgs.fee,
+            orderInfo.tradeArgs.isNegativeFee
+        );
+    }
+
+    /**
+     * Returns the current price of baseMarket divided by the current price of quoteMarket. This
+     * value is multiplied by 10^18.
+     */
+    function getCurrentPrice(
+        uint256 baseMarket,
+        uint256 quoteMarket
+    )
+        private
+        view
+        returns (uint256)
+    {
+        Monetary.Price memory basePrice = SOLO_MARGIN.getMarketPrice(baseMarket);
+        Monetary.Price memory quotePrice = SOLO_MARGIN.getMarketPrice(quoteMarket);
+        return basePrice.value.mul(PRICE_BASE).div(quotePrice.value);
+    }
+
+    /**
+     * Parses the order, verifies that it is not expired or canceled, and verifies the signature.
+     */
+    function getOrderAndValidateSignature(
+        bytes memory data
+    )
+        private
+        returns (OrderInfo memory)
+    {
+        Require.that(
+            (
+                data.length == NUM_ORDER_BYTES ||
+                data.length == NUM_ORDER_BYTES + NUM_SIGNATURE_BYTES
+            ),
+            FILE,
+            "Cannot parse order from data"
+        );
+
+        // load orderInfo from calldata
+        OrderInfo memory orderInfo;
+        (
+            orderInfo.order,
+            orderInfo.tradeArgs
+        ) = abi.decode(data, (Order, TradeArgs));
+
+        // load tradeArgs from storage if price is zero
+        if (orderInfo.tradeArgs.price == 0) {
+            orderInfo.tradeArgs = g_tradeArgs;
+            g_tradeArgs = TradeArgs({
+                price: 0,
+                fee: 0,
+                isNegativeFee: false
+            });
+        }
+        Require.that(
+            orderInfo.tradeArgs.price != 0,
+            FILE,
+            "TradeArgs loaded price is zero"
+        );
+
+        orderInfo.orderHash = getOrderHash(orderInfo.order);
+        OrderStatus orderStatus = g_status[orderInfo.orderHash];
+
+        // verify valid signature or is pre-approved
```

Comment:
It's strange that some of these validations are happening in the "get" function

Reply:
Renamed and refactored a bit